### ISSUE TITLE
Docfx Error extracting Metadata from assembly: Unable to generate spec reference for !:

### ISSUE THREAD

**Issue by Herrmel** [dotnet/docfx#4952]
**Operation System**: `Windows` 

**DocFX Version Used**: 2.44, 2.43.3

**Template used**: `default` 

**Framework used**: Net Core 2.2 and 3.0

**Steps to Reproduce**:
Did not find out how to reproduce yet.

**Expected Behavior**:
Metadata is generated

**Actual Behavior**:
The following Error Occurrs:

```
Error:Error extracting metadata for <TheProjects>: Microsoft.DocAsCode.Exceptions.DocfxException: Unable to generate spec reference for !: ---> System.IO.InvalidDataException: Fail to parse id for symbol  in namespace .
   bei Microsoft.DocAsCode.Metadata.ManagedReference.YamlModelGenerator.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters, Dictionary`2 references, SymbolVisitorAdapter adapter)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters)
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.AddSpecReference(ISymbol symbol, IReadOnlyList`1 typeGenericParameters, IReadOnlyList`1 methodGenericParameters)
   bei System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   bei System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   bei System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.GenerateInheritance(INamedTypeSymbol symbol, MetadataItem item)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitNamedType(INamedTypeSymbol symbol)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitDescendants[T](IEnumerable`1 children, Func`2 getChildren, Func`2 filter)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitNamespace(INamespaceSymbol symbol)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitDescendants[T](IEnumerable`1 children, Func`2 getChildren, Func`2 filter)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.SymbolVisitorAdapter.VisitAssembly(IAssemblySymbol symbol)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.RoslynMetadataExtractor.Extract(ExtractMetadataOptions options)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.GetMetadataFromProjectLevelCache(IBuildController controller, IInputParameters key)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.<SaveAllMembersFromCacheAsync>d__13.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Microsoft.DocAsCode.Metadata.ManagedReference.ExtractMetadataWorker.<ExtractMetadataAsync>d__11.MoveNext()
```

When debugging Docfx I get a lot of failed Debug assertions for Enums:
![image](https://user-images.githubusercontent.com/36636591/62357662-d25fa380-b513-11e9-8a60-9bf05e51b91f.png)
In CSYamlModelGenerator.GetLiteralExpression I can see that the Parameter `ITypeSymbol type` shuold be an enum. But `type.TypeKind` is `TypeKind.Class` instead of `TypeKind.Enum`. When building only the project with those Enums the Assertions come up but it won't crash so I'm not 100% sure its related to the Error. But I could reproduce this part by just creating a library with one simple enum but here I sometimes got a new failed Assertion: System.Object could not be resolved in `ResolveReference.TryAddReference`. But i dont get those in my original projects so whatever :D

In `YamlModelGenerator.AddSpecReference` and `SymbolVisitorAdapter.AddSpecReference` (where the crucial error occurrs) the Parameter `symbol` is `Type.Error` and overall pretty empty but I don't know where its coming from. The Parameter `references` also contain some Enums so maybe thats why? I could not find a way to reproduce this part outside of the projects yet.

Thanks for your help.

**Comment by Herrmel** [comment]
It seems I fixed the error by upgrading all nuget-packeges from roslyn to the newest version.(Microsoft.CodeAnalysis.*) I also had to add a new Nuget to `Microsoft.DocAsCode.Metadata.ManagedReference.Roslyn`(`Microsoft.CodeAnalysis.Workspaces.MSBuild`), probably just got outsourced in a new own nuget. But with my selfbuild version in the Buildstep i get following warnings and only markdown is processed:
```
Warning:[ExportResourceFiles]No resource found for [default].
Warning:[BuildCore.Build Document.LinkPhaseHandlerWithIncremental.Apply Templates]There is no template processing document type(s): Conceptual,ManagedReference,Toc
```
It somehow can't find the default template? But building the now generated metadata with an older version works clean.

**Comment by superyyrrzz** [comment]
Could you provide some source to help reproduce this issue?

**Comment by Herrmel** [comment]
For the Enumproblem its every enum.
this would be an example for a lib with one simple enum:[EnumProblem.zip](https://github.com/dotnet/docfx/files/3480700/EnumProblem.zip)

But for the Error I could not really find a way to reproduce this in a smaller project just the Fix for it. I'll go on and try to figure it out maybe I'll find something

**Comment by Herrmel** [comment]
I also get the following failed Assertion I already mentioned by simply adding a class to the [Library](https://github.com/dotnet/docfx/files/3480929/TestLibrary2.zip):
![image](https://user-images.githubusercontent.com/36636591/62692222-8e1b4a00-b9d0-11e9-988d-cf469e1eb4b6.png)

**Comment by Herrmel** [comment]
Ok I finally found it. It happens when a Class implements a GenericInterface with a Tuple that has named values like this:
```cs
public class SupertestClass : GenericInterface<(string arg1,string arg2)>
```
your current version of roslyn probably wasnt capeable of handling the names "arg1" and "arg2" and declared the interface as `Typekind.ErrorKind`.
Heres the library for reference: [TestLibraryFinal.zip](https://github.com/dotnet/docfx/files/3481498/TestLibraryFinal.zip)

**Comment by superyyrrzz** [comment]
related issue: #3211

**Comment by matkoch** [comment]
@superyyrrzz 	can you please prioritize this? The related issue is already a year old, and this is more or less the reason I couldn't update my API docs since then.

**Comment by Herrmel** [comment]
At least the enum problem would be nice. Looks kinda odd in the documentation. The other problems I can fix by simply upgrading roslyn-packages and compile it myself.
Btw what is holding you back from upgrading roslyn in your release version? It worked really well for me.

**Comment by superyyrrzz** [comment]
@Herrmel My previous changes just make it work with VS2019. I did not try upgrading Roslyn. Of course PR is welcomed 😄 

@matkoch Sorry for the delay fix as [we now have limited resource on docfx v2](https://github.com/dotnet/docfx/blob/dev/Roadmap.md#v2-strategy).

**Comment by Herrmel** [comment]
@superyyrrzz So I just tested the new version of docfx and pretty much nothing changed for me. Could you provide the docfx config that made it work for you? For the Tuples its enough for me to upgrade Microsoft.CodeAnalysis again to newest version. The enums Symbol.TypeKind however is not `TypeKind.Enum', the debugassertion fails and is badly formatted on the website output. Could this be a Roslyn related problem? Or do I need to add a Reference to the Enumclass with `references` like in the related issue?

**Comment by superyyrrzz** [comment]
@Herrmel Sorry for the confusion statement before. I mean there is a bug before that docfx cannot work on machines installed with VS2019. My previous changes fixed this bug, but did not test anything related to this bug, and not upgrade Roslyn version.

It may need some code change to format the output value. I don't this it needs a `references` as it is from BCL.

**Comment by dlech** [comment]
Microsoft.CodeAnalysis packages are updated in #5163 so this may be fixed now.

**Comment by Herrmel** [comment]
Sorry for the long time to respond. So I tested the new Docfx Version and indeed the Errors for tuples are gone, thank you.
But [the Problem here](https://github.com/dotnet/roslyn/issues/38816) still persist. 
For Example this affects the results of docfx for enums so it does not use the enum Template but instead renders it as a class inheriting from Object(not even System.Enum) with the underlying implementation i guess(all declared Enums are constant fields plus an additional field `value__` with the actual valuetype).

Also all References between dll's are not in the output(e.g. types that inherit from other dll inherit Object).

Should I open a new Issue for this here since this topic was closed? Anyway I guess we have to wait for roslyn to finally fix this.

BTW how is docfx v3 doing?^^

**Comment by Herrmel** [comment]
ok nevermind. I found out how to get it working for me. For those having similar Problems:
I removed one line in docfx(In class `CompilationUtility`, Method `CreateCompilationFromAssembly`  the line `paths.Add(typeof(object).Assembly.Location);`) since .Net Core can not use the NetFramework assembly.
Then I added References for the NetCore dlls in docfx.json like this:
```json
"references": [
	{
		"files": [
			"mscorlib.dll",
			"System*.dll"
		],
		"src": "C:/Program Files/dotnet/shared/Microsoft.NETCore.App/3.1.1/"
	}
],
```
